pkg/descheduler/framework/plugins/loadaware/utilization_util.go

```diff
@@ -243,40 +312,114 @@ func evictPodsFromSourceNodes(
 	continueEviction continueEvictionCond,
 	evictionReasonGenerator evictionReasonGeneratorFn,
 ) {
-	var targetNodes []*corev1.Node
-	totalAvailableUsages := map[corev1.ResourceName]*resource.Quantity{}
-	for _, destinationNode := range destinationNodes {
-		targetNodes = append(targetNodes, destinationNode.node)
+	totalAvailableUsages, targetNodes := targetAvailableUsage(destinationNodes, resourceNames, false)
+	prodAvailableUsages, prodTargetNodes := targetAvailableUsage(prodDestinationNodes, resourceNames, true)
+	bothTotalAvailableUsage, bothTotalNodes := targetAvailableUsage(bothDestinationNodes, resourceNames, false)
+	prodBothAvailableUsage, prodBothTotalNodes := targetAvailableUsage(bothDestinationNodes, resourceNames, true)
+	klog.V(4).InfoS("node pool availableUsage", "onlyNodeTotal", totalAvailableUsages, "onlyProdOnly", prodAvailableUsages,
+		"bothLowNodesTotal", bothTotalAvailableUsage, "bothLowProdTotal", prodBothAvailableUsage)
+
+	nodeTotalAvailableUsages := newAvailableUsage(resourceNames)
+	for _, resourceName := range resourceNames {
+		if quantity, ok := nodeTotalAvailableUsages[resourceName]; ok {
+			if _, totalOk := totalAvailableUsages[resourceName]; totalOk {
+				quantity.Add(*totalAvailableUsages[resourceName])
+			}
+			if _, bothOk := bothTotalAvailableUsage[resourceName]; bothOk {
+				quantity.Add(*bothTotalAvailableUsage[resourceName])
+			}
+		}
+	}
+	nodeKeysAndValues := []interface{}{
+		"nodePool", nodePoolName,
+	}
+	for resourceName, quantity := range nodeTotalAvailableUsages {
+		nodeKeysAndValues = append(nodeKeysAndValues, string(resourceName), quantity.String())
+	}
+	klog.V(4).InfoS("Total node usage capacity to be moved", nodeKeysAndValues...)
+
+	targetNodes = append(targetNodes, bothTotalNodes...)
+	balancePods(ctx, nodePoolName, sourceNodes, targetNodes, nodeTotalAvailableUsages, dryRun, nodeFit, false, resourceWeights, podEvictor,
+		podFilter, nodeIndexer, continueEviction, evictionReasonGenerator)
+
+	// bothLowNode will be used by nodeHigh and prodHigh nodes, needs sub resources used by pods on nodeHigh.
+	for _, resourceName := range resourceNames {
+		if quantity, ok := nodeTotalAvailableUsages[resourceName]; ok {
+			// A part of bothTotalAvailableUsage has been used,
+			// then the remaining part of nodeTotalAvailableUsage can be utilized.
+			if bothTotalAvailableUsage[resourceName].Cmp(*quantity) > 1 {
```

Comment:
I think it should be `if bothTotalAvailableUsage[resourceName].Cmp(*quantity) > 0`. Check the definition of Cmp function?